File: flink-ai-flow/ai_flow/model_center/entity/registered_model.py
#
# Licensed to the Apache Software Foundation (ASF) under one
# or more contributor license agreements.  See the NOTICE file
# distributed with this work for additional information
# regarding copyright ownership.  The ASF licenses this file
# to you under the Apache License, Version 2.0 (the
# "License"); you may not use this file except in compliance
# with the License.  You may obtain a copy of the License at
#
#   http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing,
# software distributed under the License is distributed on an
# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
# KIND, either express or implied.  See the License for the
# specific language governing permissions and limitations
# under the License.
#
from ai_flow.model_center.entity._model_repo_entity import _ModelRepoEntity


class RegisteredModel(_ModelRepoEntity):
    """
    AIFlow entity for Registered Model.
    A registered model entity is uniquely identified by its name.
    """

    def __init__(self, model_name):
        """
        Construct a :py:class:`ai_flow.model_center.entity.RegisteredModel`

        :param model_name: Unique name for this registered model within Model Registry.
        """
        super(RegisteredModel, self).__init__()
        self._model_name = model_name

    @property
    def model_name(self):
        """String. Unique name for this registered model within Model Registry."""
        return self._model_name

    # proto mappers
    @classmethod
    def from_proto(cls, proto):
        model_meta = proto.model_meta
        return cls(model_meta.model_name.value if model_meta.HasField("model_name") else None)

    @classmethod
    def from_resp_proto(cls, proto):
        return cls(proto.model_name.value if proto.HasField("model_name") else None)

File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/entity/RegisteredModel.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package com.aiflow.entity;

import com.aiflow.proto.Message.ModelMetaParam;
import com.aiflow.proto.Message.RegisteredModelDetail;
import com.aiflow.proto.Message.RegisteredModelMeta;
import com.aiflow.proto.Message.RegisteredModelMetas;

import java.util.ArrayList;
import java.util.List;

public class RegisteredModel {

    private String modelName;
    private String modelDesc;
    private ModelVersion latestModelVersion;

    public RegisteredModel() {
    }

    public RegisteredModel(String modelName) {
        this.modelName = modelName;
    }

    public RegisteredModel(String modelName, String modelDesc) {
        this.modelName = modelName;
        this.modelDesc = modelDesc;
    }

    public RegisteredModel(String modelName, String modelDesc, ModelVersion latestModelVersion) {
        this.modelName = modelName;
        this.modelDesc = modelDesc;
        this.latestModelVersion = latestModelVersion;
    }

    public String getModelName() {
        return modelName;
    }

    public void setModelName(String modelName) {
        this.modelName = modelName;
    }

    public String getModelDesc() {
        return modelDesc;
    }

    public void setModelDesc(String modelDesc) {
        this.modelDesc = modelDesc;
    }

    public ModelVersion getLatestModelVersion() {
        return latestModelVersion;
    }

    public void setLatestModelVersion(ModelVersion latestModelVersion) {
        this.latestModelVersion = latestModelVersion;
    }

    @Override
    public String toString() {
        return "RegisteredModel{" +
                "modelName='" + modelName + '\'' +
                ", modelDesc='" + modelDesc + '\'' +
                ", latestModelVersion=" + latestModelVersion +
                '}';
    }

    public static RegisteredModel buildRegisteredModel(ModelMetaParam modelMetaParam) {
        return modelMetaParam == null ? null : new RegisteredModel(modelMetaParam.getModelName().getValue());
    }

    public static RegisteredModel buildRegisteredModel(RegisteredModelMeta registeredModelMeta) {
        return registeredModelMeta == null ? null : new RegisteredModel(registeredModelMeta.getModelName(),
                registeredModelMeta.getModelDesc().getValue());
    }

    public static RegisteredModel buildRegisteredModel(RegisteredModelDetail registeredModelDetail) {
        return registeredModelDetail == null ? null : new RegisteredModel(registeredModelDetail.getRegisteredModel().getModelName(),
                registeredModelDetail.getRegisteredModel().getModelDesc().getValue(),
                ModelVersion.buildModelVersion(registeredModelDetail.getModelVersion()));
    }

    public static List<RegisteredModel> buildRegisteredModels(RegisteredModelMetas registeredModelMetas) {
        if (registeredModelMetas == null) {
            return null;
        } else {
            List<RegisteredModel> registeredModels = new ArrayList<>();
            for (RegisteredModelMeta registeredModelMeta : registeredModelMetas.getRegisteredModelsList()) {
                registeredModels.add(buildRegisteredModel(registeredModelMeta));
            }
            return registeredModels;
        }
    }
}